src/mpi/coll/allreduce/allreduce_intra_k_reduce_scatter_allgather.c

```diff
@@ -36,6 +36,8 @@ int MPIR_Allreduce_intra_k_reduce_scatter_allgather(const void *sendbuf,
     MPIR_CHKLMEM_DECL(2);
 
     MPIR_Assert(k > 1);
+    /* This algorithm uses cached data in comm, thus it won't work with coll_group */
+    MPIR_Assert(coll_group == MPIR_SUBGROUP_NONE);
```

Comment:
are we losing the ability to use these algorithms on node and node roots groups? is it significant?

Reply:
No. But we need either not rely on cached data or some caching scheme that differentiates group hierarchy. For now, left as TODO.